tests/topotests/multicast_pim6_sm_topo1/test_multicast_pim6_sm2.py

```diff
@@ -0,0 +1,621 @@
+# DAMAGES WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS,
+# WHETHER IN AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS
+# ACTION, ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE
+# OF THIS SOFTWARE.
+#
+
+"""
+Following tests are covered to test multicast pim6 sm:
+
+Test steps
+- Create topology (setup module)
+- Bring up topology
+
+Following tests are covered:
+1. Verify (*,G) and (S,G) entry populated again after clear the
+PIM nbr and mroute from FRR node
+2. Verify SPT switchover working when RPT and SPT path is
+different
+"""
+
+import os
+import sys
+import json
+import time
+import datetime
+import pytest
+
+# Save the Current Working Directory to find configuration files.
+CWD = os.path.dirname(os.path.realpath(__file__))
+sys.path.append(os.path.join(CWD, "../"))
+sys.path.append(os.path.join(CWD, "../lib/"))
+
+# Required to instantiate the topology builder class.
+
+# pylint: disable=C0413
+# Import topogen and topotest helpers
+from lib.topogen import Topogen, get_topogen
+
+from lib.common_config import (
+    start_topology,
+    write_test_header,
+    write_test_footer,
+    step,
+    reset_config_on_routers,
+    shutdown_bringup_interface,
+    start_router,
+    stop_router,
+    create_static_routes,
+    required_linux_kernel_version,
+    socat_send_mld_join,
+    socat_send_pim6_traffic,
+    get_frr_ipv6_linklocal,
+)
+from lib.bgp import create_router_bgp
+from lib.pim import (
+    create_pim_config,
+    create_mld_config,
+    verify_mld_groups,
+    verify_mroutes,
+    clear_pim6_interface_traffic,
+    verify_upstream_iif,
+    clear_pim6_mroute,
+    verify_pim_interface_traffic,
+    verify_pim_state,
+    McastTesterHelper,
+    verify_pim_join,
+    verify_mroute_summary,
+    verify_pim_nexthop,
+    verify_sg_traffic,
+    verify_mld_config,
+)
+
+from lib.topolog import logger
+from lib.topojson import build_config_from_json
+
+# Global variables
+GROUP_RANGE = "ff00::/8"
+
+GROUP_RANGE_1 = [
+    "ffaa::1/128",
+    "ffaa::2/128",
+    "ffaa::3/128",
+    "ffaa::4/128",
+    "ffaa::5/128",
+]
+MLD_JOIN_RANGE_1 = ["ffaa::1", "ffaa::2", "ffaa::3", "ffaa::4", "ffaa::5"]
+
+GROUP_RANGE_2 = [
+    "ffbb::1/128",
+    "ffbb::2/128",
+    "ffbb::3/128",
+    "ffbb::4/128",
+    "ffbb::5/128",
+]
+MLD_JOIN_RANGE_2 = ["ffbb::1", "ffbb::2", "ffbb::3", "ffbb::4", "ffbb::5"]
+GROUP_RANGE_3 = [
+    "ffcc::1/128",
+    "ffcc::2/128",
+    "ffcc::3/128",
+    "ffcc::4/128",
+    "ffcc::5/128",
+]
+MLD_JOIN_RANGE_3 = ["ffcc::1", "ffcc::2", "ffcc::3", "ffcc::4", "ffcc::5"]
+
+HELLO_TIMER = 1
+HOLD_TIMER = 3
+PREFERRED_NEXT_HOP = "link_local"
+ASSERT_MSG = "Testcase {} : Failed Error: {}"
+
+pytestmark = [pytest.mark.pim6d]
+
+
+def setup_module(mod):
+    """
+    Sets up the pytest environment
+
+    * `mod`: module name
+    """
+
+    # Required linux kernel version for this suite to run.
+    result = required_linux_kernel_version("4.19")
```

Comment:
Is it required to strict version for these tests?

Reply:
We are using this validation for multicast routing, and this information was provided to us by community and same information is updated in "FRR User Manual, Release latest" table.